backend/fms_core/resources/extraction.py

```diff
@@ -126,25 +124,29 @@ def import_field(self, field, obj, data, is_m2m=False):
                     comment=f"Automatically generated via extraction template import on "
                             f"{datetime.utcnow().isoformat()}Z"
                 )
+        else:
+            parent = None
+            self.row_warnings.append(f"Parent rack container will not be created if you do not provide [Nucleic Acid Location Barcode, Nucleic Acid Location Coord].")
+
+        # Per Alex: We can make new tubes if needed for extractions
+
+        # Information that can be used to either retrieve or create a new
+        # tube container. It is of type tube specifically because, as
+        # mentioned above, extractions currently only occur into 8x12 tube
+        # racks.
+        shared_container_info = dict(
+            barcode=get_normalized_str(data, "Nucleic Acid Container Barcode"),
+            # TODO: Currently can only extract into tubes
+            #  - otherwise this logic will fall apart
+            kind=CONTAINER_SPEC_TUBE.container_kind_id,
+            location=parent,
+            coordinates=get_normalized_str(data, "Nucleic Acid Location Coord"),
+        )
 
-            # Per Alex: We can make new tubes if needed for extractions
-
-            # Information that can be used to either retrieve or create a new
-            # tube container. It is of type tube specifically because, as
-            # mentioned above, extractions currently only occur into 8x12 tube
-            # racks.
-            shared_container_info = dict(
-                barcode=get_normalized_str(data, "Nucleic Acid Container Barcode"),
-                # TODO: Currently can only extract into tubes
-                #  - otherwise this logic will fall apart
-                kind=CONTAINER_SPEC_TUBE.container_kind_id,
-                location=parent,
-                coordinates=get_normalized_str(data, "Nucleic Acid Location Coord"),
-            )
-
+        try:
+            obj.container = Container.objects.get(**shared_container_info)
+        except Container.DoesNotExist:
```

Comment:
I recommend using that approach, much more concise: https://docs.djangoproject.com/en/3.2/ref/models/querysets/#get-or-create (read the part about defaults which uses attributes only for the create)